### PR TITLE
Add dependency instructions

### PR DESCRIPTION
**Pull request by thavelick**
I added instructions for installing development dependencies for RedHat and Debian-based systems.

This is pretty complete, but I don't necessarily like how the instructions are kind of out of order now; but I didn't want to change the existing flow too much.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,20 +1,25 @@
 We love contributions from the community! Here's a quick guide to making a pull
 request:
 
+Overview
+========
+
 1. Fork the repo.
 
-2. Run the tests. We only take pull requests with passing tests, and it's great
+2. Install [Dependencies](#installing-dependencies).
```
Comment (mcmire) [inline CONTRIBUTING.md]: Maybe lowercase Dependencies?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,4 +27,38 @@ code to fit with our [code style] or the project style, or discuss alternate
...
+=======================
+
+On Debian/Ubuntu based systems
+------------------------------
+
+```
+sudo apt-get install -y ruby-dev libpq-dev libsqlite3-dev nodejs
+```
+
+Ubuntu, as of 14.04 ships with ruby 1.92. shoulda-matchers requires 2.x, so [install rvm], then install ruby-2.2:
```
Comment (mcmire) [inline CONTRIBUTING.md]: Is it possible to recommend rbenv in addition to RVM? Perhaps something like:

"shoulda-matchers requires Ruby 2.x, so use your Ruby version manager of choice to install the latest version of Ruby (2.2.1 at the time of this writing)."

Comment (mcmire) [inline CONTRIBUTING.md]: Also, perhaps:

"Ubuntu, as of 14.04, ships with Ruby 1.9.2..."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -22,4 +27,38 @@ code to fit with our [code style] or the project style, or discuss alternate
 ways of addressing the issue in question. When we're happy with everything,
 we'll bring your changes into master. Now you're a contributor!
 
+Installing Dependencies
+=======================
+
+On Debian/Ubuntu based systems
```
Comment (mcmire) [inline CONTRIBUTING.md]: "Debian/Ubuntu-based" (I know, grammar Nazi :cop:)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -22,4 +27,38 @@ code to fit with our [code style] or the project style, or discuss alternate
...
+```
+
+On RedHat-based systems
+-----------------------
+
+```
+sudo yum install -y ruby-devel postgresql-devel sqlite-devel zlib-devel
+```
+
+Also, install one of the javascript runtimes supported by [execjs]. For instance, to install node.js:
```
Comment (mcmire) [inline CONTRIBUTING.md]: JavaScript

**Comment by mcmire** [comment]
Thanks for the PR. I think the order is fine, I just had a few minor comments.

**Comment by thavelick** [comment]
Cool, I should get these changes in today or tomorrow

**Comment by thavelick** [comment]
Alright, all of your suggested edits sounded good to me and have been made. Thanks for taking a look.

**Comment by mcmire** [comment]
Thanks! Merged as 8efe20a0c4c2f5d296bd933b15fc62262f9da0e3.

**Pull request closed without merging**